e2e/charts/e2e-resources/templates/resources.yaml

```diff
@@ -0,0 +1,129 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ .Release.Name }}-container-creating
+spec:
+  containers:
+  - name: alpine
+    image: alpine:latest
+    resources:
+      limits:
+        memory: 5M
+        cpu: 5m
+      requests:
+        memory: 5M
+        cpu: 5m
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ .Release.Name }}-pending
+spec:
+  containers:
+    - name: alpine
+      image: alpine:latest
+  nodeSelector:
+    not-existing: not-existing
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Release.Name }}-stateful-set
+  labels:
+    app: statefulset
+spec:
+  selector:
+    app: statefulset
+  ports:
+    - port: 8089
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Release.Name }}-stateful-set
+spec:
+  serviceName: statefulset
+  replicas: 2
+  selector:
+    matchLabels:
+      app: statefulset
+  template:
+    metadata:
+      labels:
+        app: statefulset
+    spec:
+      containers:
+        - name: busybox
+          image: busybox
+          args:
+            - sleep
+            - infinity
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: {{ .Release.Name }}
+spec:
+  accessModes:
+    - ReadWriteOnce
+  capacity:
+    storage: 5M
+  hostPath:
+    path: /data/pv0001/
```

Comment:
Some envs might not have a PV provisioner. Could we guard this with {{- if .Values.persistentVolume.enabled }}`? I'd default it to false and `--set` it to true in the minikube-specific spec file.